cluster/aws/util.sh

```diff
@@ -733,11 +736,13 @@ function upload-server-tars() {
     SERVER_BINARY_TAR_URL=$PRESET_SERVER_BINARY_TAR_URL
     BOOTSTRAP_SCRIPT_URL=$PRESET_BOOTSTRAP_SCRIPT_URL
 
-    scp -o 'StrictHostKeyChecking no' -i ${ACCESS_FILE} ${SERVER_BINARY_TAR} ${BOOTSTRAP_SCRIPT} ${SSH_USER}@${KUBE_MASTER_IP}:/tmp
-    scp -o 'StrictHostKeyChecking no' -i ${ACCESS_FILE} ${SERVER_BINARY_TAR} ${BOOTSTRAP_SCRIPT} ${SSH_USER}@${KUBE_MINION1_IP}:/tmp
-    if [[ -n ${KUBE_MINION2_IP:-} ]]; then
-      scp -o 'StrictHostKeyChecking no' -i ${ACCESS_FILE} ${SERVER_BINARY_TAR} ${BOOTSTRAP_SCRIPT} ${SSH_USER}@${KUBE_MINION2_IP}:/tmp
-    fi    
+    if [[ $IS_PRESET_INSTANCES_DRY_RUN == $FALSE ]]; then
+      scp -o 'StrictHostKeyChecking no' -i ${ACCESS_FILE} ${SERVER_BINARY_TAR} ${BOOTSTRAP_SCRIPT} ${SSH_USER}@${KUBE_MASTER_IP}:/tmp
```

Comment:
Can scp be avoided in non dry run case if we have Jenkins CI workflow set up?

Reply:
yes, that's designed to be. In Jenkins pipeline, $IS_PRESET_INSTANCES_DRY_RUN will be true, by condition in the code, scp will be skipped. And since there is no access file in Jenkins, scp won't success anyway.